inc/sitemaps/class-sitemaps-cache.php

```diff
@@ -182,14 +423,18 @@ public static function register_clear_on_option_update( $option, $type = '' ) {
 	 *
 	 * @param string $option The option name that's being updated.
 	 *
-	 * @return string|bool Query result.
+	 * @return void
 	 */
 	public static function clear_on_option_update( $option ) {
 
-		if ( ! empty( self::$cache_clear[ $option ] ) ) {
-			return WPSEO_Sitemaps_Cache::invalidate( self::$cache_clear[ $option ] );
-		}
+		if ( array_key_exists( $option, self::$cache_clear ) ) {
+			$types = array();
 
-		return WPSEO_Sitemaps_Cache::clear();
+			if ( ! empty( self::$cache_clear[ $option ] ) ) {
+				$types[] = self::$cache_clear[ $option ];
+			}
+
+			self::clear( $types );
```

Comment:
If you change this conditional to an if-else construction it is more clear what happens.